forest/src/daemon.rs

```diff
@@ -326,9 +326,16 @@ pub(super) async fn start(config: Config, detached: bool) {
         () = sync_from_snapshot(&config, &state_manager).fuse() => {},
         _ = ctrlc_oneshot => {
             // Cancel all async services
-            for handle in services {
-                handle.cancel().await;
+            for handle in &services {
+                handle.abort();
+                // handle.is_cancelled();
             }
+
+            // TODO: confirm if we need this
+            // for handle in services {
+            //     assert!(handle.await.unwrap_err().is_cancelled());
+            // }
```

Comment:
Let's use a [JoinSet](https://docs.rs/tokio/latest/tokio/task/struct.JoinSet.html#) which will abort the tasks on drop.

Reply:
Can use `services.shutdown().await` to wait for the services to terminate and make sure the DB handle is properly dropped.

Reply:
Yep, making the change. Only thing is if the tasks panics, it would ignore them (docs say that). Are we good with that?